src/coreclr/jit/flowgraph.cpp

```diff
@@ -4988,7 +4987,7 @@ bool FlowGraphNaturalLoop::AnalyzeIteration(NaturalLoopIterInfo* info)
         return false;
     }
 
-    assert(ContainsBlock(cond->GetTrueTarget()) != ContainsBlock(cond->GetFalseTarget()));
+    // assert(ContainsBlock(cond->GetTrueTarget()) != ContainsBlock(cond->GetFalseTarget()));
```

Comment:
This is a correctness issue, but I have a local change that should make things work out even with your change here. Hopefully I'll get that in early next week.

Reply:
Thanks! No rush on that